esmlab/utils/time.py

```diff
@@ -52,109 +39,171 @@ def time_attrs(self):
         attrs = self.time.attrs
         encoding = self.time.encoding
 
-        if "units" in attrs:
-            units = attrs["units"]
-        elif "units" in encoding:
-            units = encoding["units"]
+        if 'units' in attrs:
+            units = attrs['units']
+        elif 'units' in encoding:
+            units = encoding['units']
         else:
             units = None
 
-        if "calendar" in attrs:
-            calendar = attrs["calendar"]
-        elif "calendar" in encoding:
-            calendar = encoding["calendar"]
+        if 'calendar' in attrs:
+            calendar = attrs['calendar']
+        elif 'calendar' in encoding:
+            calendar = encoding['calendar']
         else:
             calendar = DEFAULT_CALENDAR
 
-        if "bounds" in attrs:
-            bounds = attrs["bounds"]
-        elif "bounds" in encoding:
-            bounds = encoding["bounds"]
+        if 'bounds' in attrs:
+            bounds = attrs['bounds']
+        elif 'bounds' in encoding:
+            bounds = encoding['bounds']
         else:
             bounds = None
 
-        return {"units": units, "calendar": calendar, "bounds": bounds}
+        key_attrs = {'units': units, 'calendar': calendar, 'bounds': bounds}
+        other_attrs = {k: v for k, v in attrs.items() if k not in key_attrs}
```

Comment:
It looks like I'm too late with this, but you don't need to construct `other_attrs`. If the goal is to use the value from `key_attrs` if it is available and otherwise failback to what is in `attrs` then

z = attrs.copy()
z.update(key_attrs)

accomplishes that. (I think I may have had the order of the updates backwards in my original comment)

Reply:
Thank you, @mnlevy1981! we can address this in a future PR.